drivers/pwm/pwm_stm32.c

```diff
@@ -816,6 +840,17 @@ static int pwm_stm32_init(const struct device *dev)
 	cfg->irq_config_func(dev);
 #endif /* CONFIG_PWM_CAPTURE */
 
+	/* set optional timing value for channels */
+	for (i = 0; i < cfg->n_timings; i++) {
+		const struct pwm_config_channel_timing *preset = &cfg->timings[i];
+		r = pwm_set(dev, preset->channel, preset->period_ns,
+				   preset->pulse_ns, preset->flags);
+		if (r < 0) {
+			LOG_ERR("PWM timing setup failed (%d)", r);
```

Comment:
Maybe printing also the index of failed timing config?

Reply:
Done